editor/modes/visual-editor/block.js

```diff
@@ -262,8 +262,15 @@ class VisualEditorBlock extends Component {
 			return;
 		}
 
-		this.props.onSelectionStart();
-		this.props.onSelect();
+		if ( event.shiftKey ) {
+			if ( ! this.props.isSelected ) {
+				this.props.onSelectionExtension();
```

Comment:
Would be nice to have a more descriptive name for `onSelectionExtension`, but I don't know what that could be. :) I know they are controversial, but I like `maybe`-prefixed names, like `maybeMultiSelect`, but that clashes with the `on`- framing. `onShiftSelection`?

Reply:
Hm, `maybe` seems strange because something always happens. I don't mind `onShiftSelection`.